lib/experimental/class-wp-navigation-fallbacks-gutenberg.php

```diff
@@ -0,0 +1,223 @@
+	 *
+	 * @return WP_Post|null the fallback Navigation Post or null.
+	 */
+	public static function get_fallback_menu() {
+
+		$fallback_sequence = array(
+			'get_most_recently_published_navigation',
+			'create_classic_menu_fallback',
+			'create_default_fallback',
+		);
+
+		// go through each of the fallbacks and return the first one that works.
+		foreach ( $fallback_sequence as $fallback ) {
+			$navigation_post = static::$fallback();
+
+			if ( $navigation_post && ! is_wp_error( $navigation_post ) ) {
+				return $navigation_post instanceof WP_Post ? $navigation_post : static::get_most_recently_published_navigation();
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Finds the most recently published `wp_navigation` Post.
+	 *
+	 * @return WP_Post|null the first non-empty Navigation or null.
+	 */
+	public static function get_most_recently_published_navigation() {
+
+		// Default to the most recently created menu.
+		$parsed_args = array(
+			'post_type'              => 'wp_navigation',
+			'no_found_rows'          => true,
+			'update_post_meta_cache' => false,
+			'update_post_term_cache' => false,
+			'order'                  => 'DESC',
+			'orderby'                => 'date',
+			'post_status'            => 'publish',
+			'posts_per_page'         => 1, // get only the most recent.
+		);
+
+		$navigation_post = new WP_Query( $parsed_args );
+
+		if ( count( $navigation_post->posts ) > 0 ) {
+			return $navigation_post->posts[0];
+		}
+
+		return null;
+	}
+
+	/**
+	 * Creates a Navigation Menu post from a Classic Menu.
+	 *
+	 * @return int|WP_Error The post ID of the default fallback menu or a WP_Error object.
+	 */
+	public static function create_classic_menu_fallback() {
+		// See if we have a classic menu.
+		$classic_nav_menu = static::get_classic_menu_fallback();
+
+		if ( ! $classic_nav_menu ) {
+			return new WP_Error( 'no_classic_menus', 'No Classic Menus found.' );
+		}
+
+		// Todo: inject as dependency.
+		$menu_converter = new WP_Menu_Conversion_Gutenberg( $classic_nav_menu );
+
+		// If we have a classic menu then convert it to blocks.
+		$classic_nav_menu_blocks = $menu_converter->convert();
+
+		if ( empty( $classic_nav_menu_blocks ) ) {
+			return new WP_Error( 'cannot_convert_classic_menu', 'Unable to convert Classic Menu to blocks.' );
+		}
+
+		// Create a new navigation menu from the classic menu.
+		$wp_insert_post_result = wp_insert_post(
+			array(
+				'post_content' => $classic_nav_menu_blocks,
+				'post_title'   => $classic_nav_menu->name,
+				'post_name'    => $classic_nav_menu->slug,
+				'post_status'  => 'publish',
+				'post_type'    => 'wp_navigation',
+			),
+			true // So that we can check whether the result is an error.
+		);
+
+		return $wp_insert_post_result;
+	}
+
+	/**
+	 * Determine the most appropriate classic navigation menu to use as a fallback.
+	 *
+	 * @return object WP_Term The classic navigation.
+	 */
+	public static function get_classic_menu_fallback() {
```

Comment:
The process of this function looks similar to `get_fallback_menu` but there you use an array to store the function names and here you call them directly. Is there value in being consistent?

Reply:
I didn't think it was possible to ensure the same type of flow given the minor variances in logic flow.

Reply:
The naming should be consistent "fallback" should be 1st or last in all names.